### ISSUE TITLE
Automatic choosing of sender's PGP key based on current folder (includes per-email choosing)

### ISSUE THREAD

**Issue by ache** [k9mail/k-9#2155]
Based on email/alias/folder. For accounts with many PGP keys. Please.

**Comment by philipwhiuk** [comment]
Folder? That seems like a very weird reason to want to pick a different key. Nor is it clear what folder we'd be basing it on.

Can you explain why #942 is not sufficient for your use case and justify why there is sufficient demand for it?

**Comment by Valodim** [comment]
This is already being worked on, as part of #942. The granularity will be a key per sender address.

**Comment by ache** [comment]
>Folder? That seems like a very weird reason to want to pick a different key. Nor is it clear what folder we'd be basing it on.
In most cases email/alias autochoosing will be enough, this is just for completeness, i.e. Work1 forlder and Work2 folder, with automatic per-folder email address choosing first which allows automatic per-email choosing of PGP key next.
Sorry I miss #942
I agree that permanent (maybe per-folder) remembering always encrypt and only sign will be nice too.

**Comment by ArchangeGabriel** [comment]
>It is bothering operation, and user always need to remember which key it must choose. Mistake probability here is high.

Wait, you mean it’s easier to remember in which folder to go than which identity to use? (After #942, identity=key)

>Yes, I have several jobs and different PGP keys for each one.

Do you mean several PGP keys for the same identity? If not I still don’t see how this is an issue, I also have several PGP keys for different identities, and #942 solves the case where different identities are related to the same IMAP/POP.

**Comment by ArchangeGabriel** [comment]
@ache @Valodim want to keep #942 for PGP keys per identity, anything folder related should then be here. You can rename this one to “Add support for per folder identity/PGP key” and reopen it if you want, but it’s likely to be closed again because only #942 will probably be supported.

**Comment by philipwhiuk** [comment]
If you want this, the right solution is likely an issue called 'Auto-select identity based on current folder'. But it's likely to be very low priority I'd have thought given that it isn't a common workflow.

**Comment by ache** [comment]
@ArchangeGabriel

> Wait, you mean it’s easier to remember in which folder to go than which identity to use?

Yes, I use descriptive folder names.

> Do you mean several PGP keys for the same identity?

No, the same mailbox (Gmail), but different addresses inside it (they can be added in the web Gmail UI) and different PGP keys for them.

> You can rename this one to “Add support for per folder identity/PGP key” and reopen it if you want, but it’s likely to be closed again because only #942 will probably be supported.

I fully support per-email key choosing, but prefer general, much flexible approach, not only per-email but per-folder too.

**Comment by ArchangeGabriel** [comment]
OK, I can see a point of folder names being more descriptive than email addresses. But that’s still a very corner-case usage IMHO. Anyway, once #942 landed @philipwhiuk is right that this comes down to selecting the identity depending on the folder (and regarding PGP defaults w.r.t. encrypting/signing, setting it per identity should do it too).

**Comment by ache** [comment]
@ArchangeGabriel 

I am glad we come to terms.

> and reopen it

I can't change Closed label((

**Comment by saintger** [comment]
If I understand correctly, this issue would allow to specify a default sender's identity based on the current folder ?
It would be very useful. In my case I have accumulated several emails which have been used to subscribed to different marketing services.
Today I would like to gather all these emails (through forwarding) to a single folder of my main email account (main@example.com) and to reply to these emails with a single sender's alias/identity (marketing@example.com).
It is currently not possible with k9mail without selecting the identity manually each time.
So I would be very interested by this feature.
I think no other email app offer this possiblity ?

**Comment by deviantollam** [comment]
is there ANY traction yet on giving users a way to send/sign  email using different private PGP keys?  i have 7 sender identities, some are personal and some are business.

i should not be using my personal PGP key to sign email that is going out via my professional email address (which is a sender identity)

the old way was FAR superior... there was a "cryptography" setting for each the sender identity in the list.  now that's gone for reasons no one can really defend.  :-(

**Comment by deutrino** [comment]
I sort incoming mail to multiple addresses into folders on my server, and having compose & reply auto-select an identity on a per-folder basis* would solve my biggest pain point with this app. I have been badly burned more than once by composing / replying to emails with the wrong identity because K9 silently puts a default in which does not correspond to how it works in my webmail (and in the past, in thunderbird).

BTW, I don't use PGP integration, so I'm not entirely sure I'm correct, but it looks like in one comment upthread, it was suggested that PGP discussion belonged in #942 and that this issue is purely for identities. If that's the case, let's keep PGP discussion from muddying the waters here for the large number of users who don't use it.

* (I envision the reply function perhaps falling back to setting identity based on folder if a more "trustworthy" rule can't be applied - e.g. select identity based on the to: field of the message being replied to, but if that doesn't work, fall back to selecting based on folder if set).
